Geometry_Engine/Query/HashArray.cs

```diff
@@ -66,100 +86,173 @@ public static double[] IHashArray(this IGeometry igeometry)
         [Description("The geometry hash of a Curve is obtained by first retrieving any Sub-part of the curve, if present." +
             "The ISubParts() methods is able to return the 'primitive' curves that a curve is composed of. " +
             "The GeometryHashes are then calculated for the individual parts and concatenated.")]
-        private static double[] HashArray(this ICurve curve, double translationFactor)
+        private static double[] HashArray(this ICurve curve, double translationFactor, BaseComparisonConfig comparisonConfig = null, string fullName = null)
         {
+            if (comparisonConfig?.TypeExceptions?.Any(t => typeof(ICurve).IsAssignableFrom(t)) ?? false)
+                return default;
+
+            if (curve == null)
+                return default;
+
             List<ICurve> subParts = curve.ISubParts().ToList();
 
+            if (!subParts.Any())
```

Comment:
That looks pretty complex as for invalid curves 😉 The only chances for this `if` to be hit is empty `PolyCurve` or `Polyline` with less than 2 control points...

Reply:
I had experienced this in my tests -- if we remove this some test can fail, especially the ones that generate geometry at random. Doesn't hurt to keep.